python/tests/test_c_array.py

```diff
@@ -81,7 +69,7 @@ def _export_to_c(self, *args):
 
     # Make sure that this heuristic won't result in trying to import
     # something else that has an _export_to_c method
-    with pytest.raises(TypeError, match="Can't convert object of type DataType"):
+    with pytest.raises(TypeError, match="Can't resolve ArrayBuilder"):
```

Comment:
Could we try to keep that original error message? (it seems clearer) Or is this only for the low-level c array, not for the user-facing array?